### ISSUE TITLE
[Kingfisher 5.3] Prefetcher heap corruption 

### ISSUE THREAD

**Issue by seanliu1** [onevcat/Kingfisher#1147]
### Issue Description
I use prefetcher to prefetch images. I encountered crashes due to heap corruption. The issue is not existed in Kingfisher 5.2
I can frequently reproduce this issue using UITests
 
I attached the screenshot, but it should not be valuable since it is related to heap corruption

#### What

```
malloc: Heap corruption detected, free list is damaged at 0x28191e710
*** Incorrect guard value: 4332270768
```

<img width="420" alt="Screen Shot 2019-03-27 at 12 53 02 PM" src="https://user-images.githubusercontent.com/46942150/55108039-f1df4f00-508f-11e9-8693-1c209dcf8013.png">

<img width="411" alt="Screen Shot 2019-03-27 at 12 56 12 PM" src="https://user-images.githubusercontent.com/46942150/55108077-07547900-5090-11e9-9e0f-420528bf4138.png">

**Comment by seanliu1** [comment]
I have also seen` -[NSTaggedPointerString count]: unrecognized selector sent to instance 0x8000000000000000`
Not sure if it is related to heap corruption 
<img width="409" alt="Screen Shot 2019-03-27 at 1 16 41 PM" src="https://user-images.githubusercontent.com/46942150/55109246-95c9fa00-5092-11e9-950c-287740bc99aa.png">

**Comment by onevcat** [comment]
It seems to be a regression of a thread issue in image prefetcher. I will try to give it a fix today.

**Comment by onevcat** [comment]
Should be fixed in version 5.3.1.
